### PR TITLE
Jmsteur email security protocol checker

### PR DESCRIPTION
**Pull request by jmsteur**
This script can be used to check the proper configuration of the email security protocols SPF, DKIM and DMARC for a given domain.

This script automates the retrieval of the configuration for these protocols. These protocols help ensure that the provided domain are not spoofed or rejected by email servers, which is critical for protecting your domain's reputation and improving email deliverability.

This tool checks the email security settings of a domain by verifying its SPF, DKIM, and DMARC records. It extracts relevant information from a domain or an `.eml` file, performs validation, and outputs a summary of findings, including policies and IP addresses associated with SPF records. It can also analyze the alignment of sender information for SPF and DKIM, and resolve IP locations.

### CODE REVIEW

File: security/security-design/shared-assets/email-security-dns-check/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+## Purpose of the email security DNS check
```
Comment (oheimburger) [inline security/security-design/shared-assets/email-security-dns-check/README.md:1]: Ad a disclaimer for http://ip-api.com usage conditions!

**Review by oheimburger — CHANGES_REQUESTED** [review]

File: security/security-design/shared-assets/email-security-dns-check/files/email-security-dns-check/email_security_check.sh [hunk 2]
```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+###############################################################################
+# Copyright (c) 2022, 2024, Oracle and/or its affiliates.  All rights reserved.
```
[the remark this answers is not in the document]
Comment (jmsteur) [reply security/security-design/shared-assets/email-security-dns-check/files/email-security-dns-check/email_security_check.sh]: I removed 2022,

File: security/security-design/shared-assets/email-security-dns-check/README.md (hunk 1 above)
Comment (jmsteur) [reply security/security-design/shared-assets/email-security-dns-check/README.md:1]: Added the disclaimer.

File: security/security-design/shared-assets/email-security-dns-check/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+Oracle requires that certain records are added to a customers DNS to allow Oracle's email delivery service to send emails on the customers behalf. This prevents the customers emails from being flagged as spoofed. 
+
+### Disclaimer
```
Comment (oheimburger) [inline security/security-design/shared-assets/email-security-dns-check/README.md:5]: Not bad. I miss that this service is "Free for non-commercial use."

**Review by oheimburger — CHANGES_REQUESTED** [review]

File: security/security-design/shared-assets/email-security-dns-check/README.md (hunk 3 above)
Comment (jmsteur) [reply security/security-design/shared-assets/email-security-dns-check/README.md:5]: Added the remark about free for non commercial usage.

**Review by oheimburger — DISMISSED** [review]

File: security/security-design/shared-assets/email-security-dns-check/files/email-security-dns-check/scripts/email_security_check.py [hunk 4]
```diff
@@ -0,0 +1,549 @@
+##########################################################################
+# Copyright (c) 2016, 2024, Oracle and/or its affiliates.  All rights reserved.
```
[the remark this answers is not in the document]
Comment (jmsteur) [reply security/security-design/shared-assets/email-security-dns-check/files/email-security-dns-check/scripts/email_security_check.py]: Removed 2016 from the banner.

**Comment by jmsteur** [comment]
All changes are done as recommended.

**Review by oheimburger — APPROVED** [review]

**Pull request merged**